source/ocs/ocs.ts

```diff
@@ -53,17 +134,717 @@ export async function getUser(
+  let share: Promise<OcsShare>;
+
+  try {
+    share = await promisifiedGetShare.call(connection, shareId);
+  } catch (error) {
+    let reason = error.message;
+    let statusCode = '';
+    if (error.code === 404 && error.meta && error.meta.statuscode) {
+      switch (error.meta.statuscode) {
+        case 404:
+        default:
+          reason = error.meta.message;
+          break;
+      }
+
+      statusCode = error.meta.statuscode;
+    }
+
+    share = rejectWithOcsError('Unable to get share', reason, shareId, statusCode);
+  }
+
+  return share;
+}
+
+export async function deleteShare(
+  connection: OcsConnection,
+  shareId: number | string
+): Promise<boolean> {
+  let shareDeleted: Promise<boolean>;
+
+  try {
+    shareDeleted = await promisifiedDeleteShare.call(connection, shareId);
+  } catch (error) {
+    let reason = error.message;
+    let statusCode = '';
+    if (error.code === 404 && error.meta && error.meta.statuscode) {
+      switch (error.meta.statuscode) {
+        case 404:
+          reason = 'invalid shareId or the share doesn\'t exist';
+          break;
+        default:
+          reason = error.meta.message;
+          break;
+      }
+
+      statusCode = error.meta.statuscode;
+    }
+
+    shareDeleted = rejectWithOcsError('Unable to delete share', reason, shareId, statusCode);
+  }
+
+  return shareDeleted;
+}
+
+export async function addShare(
+  connection: OcsConnection,
+  path: string,
+  shareType: OcsShareType,
+  shareWith?: string,
+  permissions?: OcsSharePermissions,
+  password?: string,
+  // publicUpload?: boolean,
+): Promise<OcsShare> {
+  let addedShare: Promise<OcsShare>;
+
+  try {
+    addedShare = await promisifiedAddShare.call(connection,
+      path,
+      shareType,
+      shareWith || '',
+      (permissions !== undefined ? permissions : OcsSharePermissions.default),
+      password || '',
+      // (publicUpload !== undefined ? publicUpload : false),
+    );
+  } catch (error) {
+    let reason = error.message;
+    let statusCode = '';
+    if (
+      (error.code === 403 || error.code === 404) &&
+      error.meta && error.meta.statuscode
+    ) {
+      switch (error.meta.statuscode) {
+        case 403:
+        case 404:
+        default:
+          reason = error.meta.message;
+          break;
+      }
+
+      statusCode = error.meta.statuscode;
+    }
+
+    addedShare = rejectWithOcsError('Unable to add share', reason, path, statusCode);
+  }
+
+  return addedShare;
+}
+
+export function editShare(
+  connection: OcsConnection,
+  shareId: number | string
+) {
+  return {
+    async permissions(permissions: OcsSharePermissions): Promise<OcsShare> {
+      return await setFieldValue(connection, shareId, 'permissions', permissions);
+    },
+
+    async password(password: string): Promise<OcsShare> {
+      return await setFieldValue(connection, shareId, 'password', password);
+    },
+
+    // async publicUpload(isPublicUpload: boolean): Promise<OcsShare> {
```

Comment:
What should we do with this one?

Reply:
Please leave it for when someone has the time/motivation to figure out why the server call doesn't behave as it should. 
It will require looking through the Nextcloud server PHP code.
As this is a method that SHOULD be implemented, I've intentionally left it in there to be filled in at a later time.

Reply:
Did you create a ticket with the nextcloud devs that you could reference here?

Reply:
Fixed~